tests/unit/test_args.py

```diff
@@ -459,3 +459,226 @@ def test_client_config_beats_config_store(self):
             client_config=Config(retries={'mode': 'standard'})
         )['client_config']
         self.assertEqual(config.retries['mode'], 'standard')
+
+    def test_does_set_global_sts_endpoint_if_in_legacy_region(self):
+        should = self.args_create._should_set_global_sts_endpoint(
+            region_name='us-west-1', endpoint_url=None, endpoint_config=None
+        )
+        self.assertTrue(should)
+
+    def test_doesnt_set_global_sts_endpoint_if_not_in_legacy_region(self):
+        should = self.args_create._should_set_global_sts_endpoint(
+            region_name='eu-south-1', endpoint_url=None, endpoint_config=None
+        )
+        self.assertFalse(should)
+
+    def test_doesnt_set_global_sts_endpoint_if_endpoint_url_is_given(self):
+        should = self.args_create._should_set_global_sts_endpoint(
+            region_name='us-west-1',
+            endpoint_url="https://my.endpoint.url.com",
+            endpoint_config=None,
+        )
+        self.assertFalse(should)
+
+    def test_doesnt_set_global_sts_endpoint_if_configured(self):
+        self.args_create._config_store.set_config_variable(
+            'sts_regional_endpoints', 'regional'
+        )
+        should = self.args_create._should_set_global_sts_endpoint(
+            region_name='us-west-1', endpoint_url=None, endpoint_config=None
+        )
+        self.assertFalse(should)
+
+
+class TestEndpointResolverBuiltins(unittest.TestCase):
+    def setUp(self):
+        event_emitter = mock.Mock(HierarchicalEmitter)
+        self.config_store = ConfigValueStore()
+        self.args_create = args.ClientArgsCreator(
+            event_emitter, None, None, None, None, self.config_store
+        )
+        self.bridge = mock.Mock(ClientEndpointBridge)
+        self.bridge.endpoint_resolver = mock.Mock()
+        self.bridge.endpoint_resolver.uses_builtin_data = True
+
+        def mock_cfgvar_fn(cfgvar):
+            return False if cfgvar == 'use_fips_endpoint' else 42
+
+        self.bridge._resolve_endpoint_variant_config_var = mock.Mock(
+            side_effect=mock_cfgvar_fn
+        )
+        self.bridge._resolve_use_dualstack_endpoint = mock.Mock(
+            return_value=False
+        )
```

Comment:
Maybe a quick comment on why we're doing this?

Reply:
No longer applicable, this has been removed.